WordPress/src/main/java/org/wordpress/android/ui/posts/PostListMainViewModel.kt

```diff
@@ -90,4 +134,89 @@ class PostListMainViewModel @Inject constructor(
             }
         }
     }
+
+    private fun getAuthorFilterItems(selection: AuthorFilterSelection): List<AuthorFilterListItemUIState> {
+        return AuthorFilterSelection.values().map { value ->
+            @ColorRes val backgroundColorRes: Int =
+                    if (selection == value) R.color.grey_lighten_30_translucent_50
+                    else R.color.transparent
+
+            when (value) {
+                ME -> AuthorFilterListItemUIState.Me(accountStore.account?.avatarUrl, backgroundColorRes)
+                EVERYONE -> AuthorFilterListItemUIState.Everyone(backgroundColorRes)
+            }
+        }
+    }
+
+    /**
+     * Only the non-null variables will be changed in the current state
+     */
+    private fun updateViewState(
+        isFabVisible: Boolean? = null,
+        isAuthorFilterVisible: Boolean? = null,
+        authorFilterSelection: AuthorFilterSelection? = null,
+        authorFilterItems: List<AuthorFilterListItemUIState>? = null
+    ) {
+        IllegalArgumentException().printStackTrace()
+        val currentState = viewState.value
```

Comment:
Won't that then throw the error in production as well as debug?

Reply:
Yeap, we are OK with crashing the app even in production if it'll let us fix an issue. We observe the crashes and hotfix them if necessary and with our current release structure, there is almost no way that it'll ever make it to production if it's not working correctly.

Reply:
That goes a little counter to a comment @malinajirka  made but I am happy to make the change as I prefer fail fast code.

Reply:
Hmm.. In that case, let's wait @malinajirka's feedback on this first. He is one of the people that pushes the fail fast approach the most in Hogwarts, so I'd be curious why he gave a different feedback before. The only thing I can think of is that if a state is valid even after unexpected state occurs and the unexpected state is not impossible without a developer error, we prefer to only fail in `DEBUG`.

An example of unexpected state not being impossible without a developer error would be if we keep an enum's value in AppPrefs and then change the current enum values which in turn would make the value saved in the AppPrefs invalid. This is something we still want to be aware of, but we don't want to crash the user's app because it's still a valid state even if unexpected.

Reply:
The comment I was referring to https://github.com/wordpress-mobile/WordPress-Android/pull/9446#discussion_r268388162

Reply:
Thanks! I think he might be thinking of the `AppPrefs` issue I mentioned there, but I might be wrong, let's see what he thinks.

Reply:
I might not be consistent even with my own comments when it comes to fail fast principle. I'm sorry for the confusion, my bad. I should have elaborated more. 

TLDR: I think using `requireNotNull(viewState.value)` is fine (I actually prefer the fail-fast approach).

Regarding #9446 (comment):
Oguz's comment summarizes my thinking pretty well.

I felt like this is the case where we can be more or less sure we'll recover from the error just by returning a default value and the user won't probably even notice something went wrong. Crashing the app just because the selected filter is reseted to default isn't probably worth it.

Moreover imagine we have two values `ME, EVERYONE`. Someone renames `ME-> CURRENT_USER` and the app starts crashing in production. What do we do? We can't just rename it back, as it might introduce the same issue for users who already have `CURRENT_USER` in their prefs. So again it just felt like it's not worth it.

Reply:
The distinction makes more sense now. Thank you for clarifying.

Resolved in e582cb2a4fd4cb6f473a027f3dd984813d14a9c5